librz/bin/bobj_process.c

```diff
@@ -93,18 +93,66 @@ RZ_IPI void rz_bin_process_cxx(RzBinObject *o, char *demangled, ut64 paddr, ut64
 	*name = ':';
 }
 
+static char *find_swift_methodname(RZ_NONNULL char *demangled) {
+	// methods can be main.Tost.deinit or main.Tost.init() -> main.Tost
+	// so we will return after second dot
+	char *dot = strchr(demangled, '.');
+	dot = dot ? strchr(dot + 1, '.') : NULL;
+	if (!dot) {
+		return NULL;
+	}
+	char *methodname = dot + 1;
+	if (RZ_STR_ISEMPTY(methodname)) {
+		return NULL;
+	}
+	return methodname;
+}
+
+static void rz_bin_process_metaclass(RzBinObject *o, RzBinSymbol *symbol) {
+	if (RZ_STR_ISEMPTY(symbol->dname)) {
+		return;
+	}
+	char *no_classname = strstr(symbol->dname, "full type metadata for ");
+	char *classname = strstr(symbol->dname, "type metadata for ");
+	if (!classname || no_classname) { // only for "type metadata for class"
+		return;
+	}
+	rz_bin_object_add_field(o, classname + sizeof("type metadata for ") - 1, symbol->dname, symbol->paddr, symbol->vaddr);
```

Comment:
```suggestion
	rz_bin_object_add_field(o, classname + strlen("type metadata for "), symbol->dname, symbol->paddr, symbol->vaddr);
```

Modern compilers knows that when they see `strlen("constant string")` they can just pre-calculate the length and just set the result of the operation.